rest/data/project.go

```diff
@@ -417,3 +448,33 @@ func (pc *MockProjectConnector) UpdateProjectRevision(projectID, revision string
 func (ac *MockProjectConnector) GetVersionsInProject(project, requester string, limit, startOrder int) ([]restModel.APIVersion, error) {
 	return nil, nil
 }
+
+func (pc *MockProjectConnector) GetProjectSettingsEvent(p *model.ProjectRef) (*model.ProjectSettingsEvent, error) {
+	hook, err := model.FindGithubHook(p.Owner, p.Repo)
```

Comment:
maybe this could just return an empty `ProjectSettingsEvent`?

Reply:
I made changes to this here: https://github.com/evergreen-ci/evergreen/pull/2926/commits/e5a50d833ae90ee2a4d7cd6a89fcd18fd59dbb41#diff-5389e3ffd4b54c3122c1f91b53c7e1c6R453

Reply:
I adjusted this function to return an empty `ProjectSettingsEvent` or an error